### ISSUE TITLE
[YSQL] Assertion Failure For CREATE TABLE of DATE / TIME Types

### ISSUE THREAD

**Issue by nocaway** [YugaByte/yugabyte-db#714]
The test case related to this bug, "yb_feature_types", is commented out.  It should be turned back on once this bug is fixed.  See file "src/postgres/src/test/regress/yb_serial_schedule".

The following error is reported during regression test run.
CREATE TABLE feature_tab_date (feature_col DATE);

```
[INFO] Running org.yb.pgsql.TestPgRegress
[ERROR] Tests run: 3, Failures: 2, Errors: 1, Skipped: 0, Time elapsed: 316.769 s <<< FAILURE! - in org.yb.pgsql.TestPgRegress
[ERROR] testPgRegress(org.yb.pgsql.TestPgRegress)  Time elapsed: 268.859 s  <<< FAILURE!
java.lang.AssertionError: pg_regress exited with error code: 1, failed tests: [yb_feature_types]
	at org.yb.pgsql.TestPgRegress.testPgRegress(TestPgRegress.java:30)

[ERROR] testPgRegress(org.yb.pgsql.TestPgRegress)  Time elapsed: 268.861 s  <<< ERROR!
org.postgresql.util.PSQLException: 
ERROR: Not found: The table does not exist: table_id: "000047c5000030008000000000006755"


[ERROR] org.yb.pgsql.TestPgRegress  Time elapsed: 316.688 s  <<< FAILURE!
java.lang.AssertionError: PostgreSQL process failed to terminate normally

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestPgRegress>BasePgSQLTest.tearDownAfter:357->BasePgSQLTest.tearDownPostgreSQL:516 PostgreSQL process failed to terminate normally
[ERROR] Errors: 
[ERROR] org.yb.pgsql.TestPgRegress.testPgRegress(org.yb.pgsql.TestPgRegress)
[ERROR]   Run 1: TestPgRegress.testPgRegress:30 pg_regress exited with error code: 1, failed tests: [yb_feature_types]
[ERROR]   Run 2: TestPgRegress>BasePgSQLTest.cleanUpAfter:349 » PSQL ERROR: Not found: The tabl...
```

**Comment by nocaway** [comment]
Assigning this to @m-iancu 
- After Posgres got the table OID, if the creation failed, it appears that DROP table is being called even though the table is not yet created.  As a result, YugaByte catalog layer reports the failure to DROP statement when it should have stay silent.
- Mihnea: please check to see if we should use "DROP IF EXIST" instead of just "DROP"

**Comment by nocaway** [comment]
Note for @mbautin 
After the raising error for non-existing table, YugaByte crashed.
The crash might be due to TWO errors are raised for ONE statement execution.

postgres# CREATE TABLE feature_tab_date (feature_col DATE);

1- First, correctly raised ERROR
ERROR: DATE is not yet supported.
After this error is raised, "DROP table feature_tab_date" was executed by the system to clean up the execution.

2- Second ERROR raised but should be silent --> crash
ERROR: Not found: The table does not exist: table_id: "000047c5000030008000000000006755"
- TestPgRegress crashes after this SECOND error due to unexpected behavior?
- Should this have been a LOG(error) instead of crashing?

**Comment by m-iancu** [comment]
Fixed in 97635f4.
